cloudflare/resource_cloudflare_page_rule_test.go

```diff
@@ -2,17 +2,124 @@ package cloudflare
 
 import (
 	"fmt"
-	"os"
-	"testing"
-
-	"reflect"
-	"regexp"
-
 	"github.com/cloudflare/cloudflare-go"
 	"github.com/hashicorp/terraform/helper/resource"
 	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"reflect"
+	"regexp"
+	"testing"
 )
 
+func TestCreatesBrowserCacheTTLIntegerValues(t *testing.T) {
+	runTestSteps(t, []resource.TestStep{
+		{
+			Config: buildPageRule(`browser_cache_ttl = 1`),
+			Check:  assertActionExists("browser_cache_ttl", "1", float64(1)),
+		},
+	})
+}
+
+func TestCreatesBrowserCacheTTLThatRespectsExistingHeaders(t *testing.T) {
+	runTestSteps(t, []resource.TestStep{
+		{
+			Config: buildPageRule(`browser_cache_ttl = 0`),
+			Check:  assertActionExists("browser_cache_ttl", "0", float64(0)),
+		},
+	})
+}
+
+func TestUpdatesBrowserCacheTTLThatRespectsExistingHeaders(t *testing.T) {
+	runTestSteps(t, []resource.TestStep{
+		{
+			Config: buildPageRule(`browser_cache_ttl = 1`),
+		},
+		{
+			Config: buildPageRule(`browser_cache_ttl = 0`),
+			Check:  assertActionExists("browser_cache_ttl", "0", float64(0)),
+		},
+	})
+}
+
+func TestDeletesBrowserCacheTTLThatRespectsExistingHeaders(t *testing.T) {
+	runTestSteps(t, []resource.TestStep{
+		{
+			Config: buildPageRule(`browser_cache_ttl = 0`),
+		},
+		{
+			Config: buildPageRule(`browser_check = "on"`),
+			Check:  assertActionNotExists("browser_cache_ttl"),
+		},
+	})
+}
+
+func buildPageRule(actions string) string {
```

Comment:
Let's swap this one for a specific configuration setup (like we do for [`testAccCheckCloudflarePageRuleConfigMinify`](https://github.com/terraform-providers/terraform-provider-cloudflare/blob/b1dbaa88a803d4b069bb77c2b6c702ba2c507253/cloudflare/resource_cloudflare_page_rule_test.go#L542-L555)). It's probably a bit more verbose but it will be explicit about what is getting setup.